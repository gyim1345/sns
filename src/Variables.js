export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
