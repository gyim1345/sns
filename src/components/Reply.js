import React, { useState } from 'react';
import { css } from '@emotion/core';

function Reply({
  posting,
  currentUser,
  indexOfCommentOnThisPosting,
  addComment,
  commentAPI,
  closeModal
}) {
  const [input, setInput] = useState('');

  const onChange = e => {
    setInput(e.target.value);
  };

  const addReplyToComment = () => {
    return addComment(
      null,
      posting.postLId,
      `@${posting.userName.substring(
        0,
        posting.userName.indexOf('@')
      )} ${input}`,
      currentUser,
      commentAPI[indexOfCommentOnThisPosting].id,
      posting.id
    );
  };

  const addReply = () => {
    addReplyToComment();
    closeModal();
  };

  return (
    <>
      <textarea
        type="text"
        value={input}
        onChange={e => onChange(e)}
        css={[textAreaCss]}
      />
      <div css={[innerButtonBoxCss]}>
        <div css={innerColumnButtonBoxCss}>
          <button type="button" onClick={addReply} css={[buttonSave]}>
            Add Reply
          </button>
          {/* <AccountSetting info={info} userInfo={userInfo} /> */}
          <button
            onClick={closeModal}
            css={[buttonCancel]}
            style={{ margin: 0 }}
          >
            close
          </button>
        </div>
      </div>
      {/* <input value={input} onChange={onChange} css={[replyCss]} />
      <button type="button" onClick={addReply} id="buttonReply">
        AddReply
      </button> */}
    </>
  );
}

const textAreaCss = css`
  border: none;
  margin-top: 10px;
  font-size: 20px;
  outline: none;
  height: 100px;
  width: 500px;
  padding-left: 15px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica,
    Arial, sans-serif;
`;

const button = css`
  background-color: transparent;
  border: 1px solid rgba(var(--d0b, 219, 219, 219), 1);
  color: rgba(var(--f07, 38, 38, 38), 1);
  font-size: 18px;
  border-radius: 50%;
  border-radius: 3px;
`;

const buttonSave = css`
  ${button}
  background-color: #0275d8;
  color: white;
  margin-right: 10px;
`;

const buttonCancel = css`
  ${button}
  color: #0275d8;
  background-color: white;
`;

const modalCss = css`
  position: absolute;
  top: 50%;
  left: 50%;
  right: auto;
  bottom: auto;
  border: 1px solid rgb(204, 204, 204);
  background: rgb(255, 255, 255);
  overflow: auto;
  border-radius: 12px;
  outline: none;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  display: flex;
  flex-direction: column;
`;

const innerColumnButtonBoxCss = css`
  height: 50px;
  display: flex;
  align-items: center;
  margin-right: 15px;
`;

const innerButtonBoxCss = css`
  display: FLEX;
  background: white;
  width: 506px;
  justify-content: flex-end;
`;

const replyCss = css`
  border: none;
  outline: none;
`;

// Reply.propTypes = {
//   currentUser: PropTypes.string,
//   posting: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
//   addComment: PropTypes.elementType,
//   commentAPI: PropTypes.oneOfType([PropTypes.array]),
//   indexOfCommentOnThisPosting: PropTypes.number
// };
export default Reply;
