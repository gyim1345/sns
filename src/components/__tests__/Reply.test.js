import React from 'react';
import { mount } from 'enzyme';
import Reply from '../Reply';
import { post, comments, currentUser, currentUserName } from '../../Variables';

describe('<Reply />', () => {
  it('renders <input>, <button>, and "AddReply"', () => {
    const wrapper = mount(
      <Reply
        posting={post}
        commentAPI={comments}
        indexOfCommentOnThisPosting={0}
        currentUser={currentUser}
      />
    );

    expect(wrapper.props().posting).toBe(post);
    expect(wrapper.props().commentAPI).toBe(comments);
    expect(wrapper.props().indexOfCommentOnThisPosting).toBe(0);
    expect(wrapper.props().currentUser).toBe(currentUser);
    expect(wrapper.html()).toMatch('<input');
    expect(wrapper.html()).toMatch('<button');
    expect(wrapper.text()).toMatch('AddReply');
    expect(wrapper).toMatchSnapshot();
  });
});
