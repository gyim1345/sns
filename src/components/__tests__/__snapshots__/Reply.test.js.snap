// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reply /> renders <input>, <button>, and "AddReply" 1`] = `
<Reply
  commentAPI={
    Array [
      Object {
        "_id": "5e90119435a2d94f9e08e35a",
        "id": 1,
        "like": Array [
          "noone@gmail.com",
        ],
        "postLId": 1,
        "replyToCommentId": null,
        "title": "title1",
        "userName": "gibong@gmail.com",
      },
    ]
  }
  currentUser="gibong@gmail.com"
  indexOfCommentOnThisPosting={0}
  posting={
    Object {
      "_id": "5e8ec52d9bb4913b5a844423",
      "id": 1,
      "imageUrl": "http://localhost:3000/static/images/1.jpg",
      "like": Array [
        "gibong@gmail.com",
      ],
      "tag": Array [
        "landscape",
      ],
      "title": "posting with id 1",
      "userName": "gibong@gmail.com",
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Reply"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="input"
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJlcGx5LmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQTZDb0IiLCJmaWxlIjoiUmVwbHkuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QsIHsgdXNlU3RhdGUgfSBmcm9tICdyZWFjdCc7XG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuXG5mdW5jdGlvbiBSZXBseSh7XG4gIHBvc3RpbmcsXG4gIGN1cnJlbnRVc2VyLFxuICBpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3RpbmcsXG4gIGFkZENvbW1lbnQsXG4gIGNvbW1lbnRBUElcbn0pIHtcbiAgY29uc3QgW2lucHV0LCBzZXRJbnB1dF0gPSB1c2VTdGF0ZSgnJyk7XG5cbiAgY29uc3Qgb25DaGFuZ2UgPSBlID0+IHtcbiAgICBzZXRJbnB1dChlLnRhcmdldC52YWx1ZSk7XG4gIH07XG5cbiAgY29uc3QgYWRkUmVwbHlUb0NvbW1lbnQgPSAoKSA9PiB7XG4gICAgcmV0dXJuIGFkZENvbW1lbnQoXG4gICAgICBudWxsLFxuICAgICAgcG9zdGluZy5wb3N0TElkLFxuICAgICAgYEAke3Bvc3RpbmcudXNlck5hbWUuc3Vic3RyaW5nKFxuICAgICAgICAwLFxuICAgICAgICBwb3N0aW5nLnVzZXJOYW1lLmluZGV4T2YoJ0AnKVxuICAgICAgKX0gJHtpbnB1dH1gLFxuICAgICAgY3VycmVudFVzZXIsXG4gICAgICBjb21tZW50QVBJW2luZGV4T2ZDb21tZW50T25UaGlzUG9zdGluZ10uaWQsXG4gICAgICBwb3N0aW5nLmlkXG4gICAgKTtcbiAgfTtcblxuICBjb25zdCBhZGRSZXBseSA9ICgpID0+IHtcbiAgICBhZGRSZXBseVRvQ29tbWVudCgpO1xuICB9O1xuXG4gIHJldHVybiAoXG4gICAgPD5cbiAgICAgIDxpbnB1dCB2YWx1ZT17aW5wdXR9IG9uQ2hhbmdlPXtvbkNoYW5nZX0gY3NzPXtbcmVwbHlDc3NdfS8+XG4gICAgICA8YnV0dG9uIHR5cGU9XCJidXR0b25cIiBvbkNsaWNrPXthZGRSZXBseX0gaWQ9XCJidXR0b25SZXBseVwiPlxuICAgICAgICBBZGRSZXBseVxuICAgICAgPC9idXR0b24+XG4gICAgPC8+XG4gICk7XG59XG5cbmNvbnN0IHJlcGx5Q3NzID0gY3NzYFxuYm9yZGVyOiBub25lO1xub3V0bGluZTogbm9uZTtcbmA7XG5cbi8vIFJlcGx5LnByb3BUeXBlcyA9IHtcbi8vICAgY3VycmVudFVzZXI6IFByb3BUeXBlcy5zdHJpbmcsXG4vLyAgIHBvc3Rpbmc6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5vYmplY3QsIFByb3BUeXBlcy5hcnJheV0pLFxuLy8gICBhZGRDb21tZW50OiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGNvbW1lbnRBUEk6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5hcnJheV0pLFxuLy8gICBpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3Rpbmc6IFByb3BUeXBlcy5udW1iZXJcbi8vIH07XG5leHBvcnQgZGVmYXVsdCBSZXBseTtcbiJdfQ== */",
        "name": "1e9bb73-replyCss-Reply",
        "next": undefined,
        "styles": "border:none;outline:none;;label:replyCss;;;label:Reply;/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJlcGx5LmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQXFDK0MiLCJmaWxlIjoiUmVwbHkuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QsIHsgdXNlU3RhdGUgfSBmcm9tICdyZWFjdCc7XG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuXG5mdW5jdGlvbiBSZXBseSh7XG4gIHBvc3RpbmcsXG4gIGN1cnJlbnRVc2VyLFxuICBpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3RpbmcsXG4gIGFkZENvbW1lbnQsXG4gIGNvbW1lbnRBUElcbn0pIHtcbiAgY29uc3QgW2lucHV0LCBzZXRJbnB1dF0gPSB1c2VTdGF0ZSgnJyk7XG5cbiAgY29uc3Qgb25DaGFuZ2UgPSBlID0+IHtcbiAgICBzZXRJbnB1dChlLnRhcmdldC52YWx1ZSk7XG4gIH07XG5cbiAgY29uc3QgYWRkUmVwbHlUb0NvbW1lbnQgPSAoKSA9PiB7XG4gICAgcmV0dXJuIGFkZENvbW1lbnQoXG4gICAgICBudWxsLFxuICAgICAgcG9zdGluZy5wb3N0TElkLFxuICAgICAgYEAke3Bvc3RpbmcudXNlck5hbWUuc3Vic3RyaW5nKFxuICAgICAgICAwLFxuICAgICAgICBwb3N0aW5nLnVzZXJOYW1lLmluZGV4T2YoJ0AnKVxuICAgICAgKX0gJHtpbnB1dH1gLFxuICAgICAgY3VycmVudFVzZXIsXG4gICAgICBjb21tZW50QVBJW2luZGV4T2ZDb21tZW50T25UaGlzUG9zdGluZ10uaWQsXG4gICAgICBwb3N0aW5nLmlkXG4gICAgKTtcbiAgfTtcblxuICBjb25zdCBhZGRSZXBseSA9ICgpID0+IHtcbiAgICBhZGRSZXBseVRvQ29tbWVudCgpO1xuICB9O1xuXG4gIHJldHVybiAoXG4gICAgPD5cbiAgICAgIDxpbnB1dCB2YWx1ZT17aW5wdXR9IG9uQ2hhbmdlPXtvbkNoYW5nZX0gY3NzPXtbcmVwbHlDc3NdfS8+XG4gICAgICA8YnV0dG9uIHR5cGU9XCJidXR0b25cIiBvbkNsaWNrPXthZGRSZXBseX0gaWQ9XCJidXR0b25SZXBseVwiPlxuICAgICAgICBBZGRSZXBseVxuICAgICAgPC9idXR0b24+XG4gICAgPC8+XG4gICk7XG59XG5cbmNvbnN0IHJlcGx5Q3NzID0gY3NzYFxuYm9yZGVyOiBub25lO1xub3V0bGluZTogbm9uZTtcbmA7XG5cbi8vIFJlcGx5LnByb3BUeXBlcyA9IHtcbi8vICAgY3VycmVudFVzZXI6IFByb3BUeXBlcy5zdHJpbmcsXG4vLyAgIHBvc3Rpbmc6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5vYmplY3QsIFByb3BUeXBlcy5hcnJheV0pLFxuLy8gICBhZGRDb21tZW50OiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGNvbW1lbnRBUEk6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5hcnJheV0pLFxuLy8gICBpbmRleE9mQ29tbWVudE9uVGhpc1Bvc3Rpbmc6IFByb3BUeXBlcy5udW1iZXJcbi8vIH07XG5leHBvcnQgZGVmYXVsdCBSZXBseTtcbiJdfQ== */",
        "toString": [Function],
      }
    }
    onChange={[Function]}
    value=""
  >
    <input
      className="css-1e9bb73-replyCss-Reply"
      onChange={[Function]}
      value=""
    />
  </EmotionCssPropInternal>
  <button
    id="buttonReply"
    onClick={[Function]}
    type="button"
  >
    AddReply
  </button>
</Reply>
`;
